### PR TITLE
chore(docs): faq - kubernetes

### PR DESCRIPTION
**Pull request by ShaiBer**
Fixes: #2189 
*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: docs/07-faq/023-kubernetes-support.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+title: Does Wing support Kebernetes?
```
Comment (eladb) [inline docs/07-faq/023-kubernetes-support.md:2]:
```suggestion
title: Does Wing support Kubernetes?
```

File: docs/07-faq/023-kubernetes-support.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+id: kubernetes-support
+keywords: [faq, supported provisioning engines, winglang, Wing programming language, Wing language, Kubernetes, K8S]
+---
+
+Not yet but we will in the future. Our roadmap is [here](https://docs.winglang.io/status#roadmap).
+
+You can add your vote for Kubernetes on [this issue](https://github.com/winglang/wing/issues/2066) to help increase its priority.
```
Comment (eladb) [inline docs/07-faq/023-kubernetes-support.md:9]: This issue is general. I think this should probably link to an issue about Kubernetes as a target specifically.

File: docs/07-faq/023-kubernetes-support.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+title: Does Wing support Kebernetes?
+id: kubernetes-support
+keywords: [faq, supported provisioning engines, winglang, Wing programming language, Wing language, Kubernetes, K8S]
+---
+
+Not yet but we will in the future. Our roadmap is [here](https://docs.winglang.io/status#roadmap).
```
Comment (eladb) [inline docs/07-faq/023-kubernetes-support.md:7]: CDK8s is supposed to be supported

**Review by eladb — CHANGES_REQUESTED** [review]

File: docs/07-faq/023-kubernetes-support.md (hunk 2 above)
Comment (ShaiBer) [reply docs/07-faq/023-kubernetes-support.md]: There is a comment there about K8S with some votes already

File: docs/07-faq/023-kubernetes-support.md (hunk 3 above)
Comment (ShaiBer) [reply docs/07-faq/023-kubernetes-support.md]: Changed to be more granular

File: docs/07-faq/023-kubernetes-support.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+title: Does Wing support Kubernetes?
+id: kubernetes-support
+keywords: [faq, supported provisioning engines, winglang, Wing programming language, Wing language, Kubernetes, K8S]
+---
+
+The short answer us that we don't support Kubernetes yet, but we will in the future. 
+Our roadmap is [here](https://docs.winglang.io/status#roadmap).
+
+The longer answer is that there are a different kinds of Kubernetes support with varying levels of maturity in Wing:
+1. Creation and configuration of K8S clusters through the [K8S Terraform Provider](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs). Since Wing supports any Terraform provider, it should work, but we habe not officially tested it yet.
```
Comment (Chriscbr) [inline docs/07-faq/023-kubernetes-support.md:11]: This should take only a few minutes to test, we can install `https://www.npmjs.com/package/@cdktf/provider-kubernetes` and see if it's possible to instantiate a few resources like a pod or deployment in Wing. I can write up an example tomorrow

```suggestion
1. Creation and configuration of K8S clusters through the [K8S Terraform Provider](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs). These resources can be used by installing `@cdktf/provider-kubernetes` from npm and using the resources in Wing:

(example code)
```

File: docs/07-faq/023-kubernetes-support.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+2. Using Kubernetes to run long-running services. This is not yet supported. You can add your vote for it on [this issue](https://github.com/winglang/wing/issues/2319) to help increase its priority.
+3. Using Kubernetes as a provisioning engine though crossplane. You can add your vote for it on [this issue](https://github.com/winglang/wing/issues/2066) to help increase its priority.
```
Comment (Chriscbr) [inline docs/07-faq/023-kubernetes-support.md:13]: WDYT of adding `Supporting kubernetes as a built-in compilation target for resources in the [Wing SDK](link)`?

File: docs/07-faq/023-kubernetes-support.md (hunk 4 above)
Comment (ShaiBer) [reply docs/07-faq/023-kubernetes-support.md:11]: Cool!

File: docs/07-faq/023-kubernetes-support.md (hunk 5 above)
Comment (ShaiBer) [reply docs/07-faq/023-kubernetes-support.md:15]: Instead of the third point?

File: docs/07-faq/023-kubernetes-support.md (hunk 4 above)
Comment (Chriscbr) [reply docs/07-faq/023-kubernetes-support.md]: OK - so I had a chance to play around with this, here's the example code in Wing:

```js
bring cloud;
bring "@cdktf/provider-kubernetes" as kube;

new kube.provider.KubernetesProvider();

let ns = new kube.namespace.Namespace(metadata: kube.namespace.NamespaceMetadata {
  name: "nginx",
});

let deployment = new kube.deployment.Deployment(metadata: kube.deployment.DeploymentMetadata {
  name: "nginx",
  namespace: ns.metadata.name,
}, spec: kube.deployment.DeploymentSpec {
  template: kube.deployment.DeploymentSpecTemplate {
    metadata: kube.deployment.DeploymentSpecTemplateMetadata {
      labels: {
        app: "nginx",
      },
    },
    spec: kube.deployment.DeploymentSpecTemplateSpec {
      container: [
        kube.deployment.DeploymentSpecTemplateSpecContainer {
          name: "nginx-container",
          image: "nginx",
          port: kube.deployment.DeploymentSpecTemplateSpecContainerPort {
            container_port: 80,
          },
        }
      ]
    }
  }
});
```

It produces valid terraform which is cool 👍 

But there are several ergonomic issues:
- IDE autocomplete doesn't work after you type `new kube.deployment.`
- You have to type the full type names everywhere, and it gets very cumbersome
- To compile you need to specify a cloud like `wing compile -t tf-aws` even though the template you produce isn't specific to any cloud, so it seems weird (I think we need another target like `tf` or `tf-base`?)

So maybe it's not worth showing off this capability in the FAQ yet

File: docs/07-faq/023-kubernetes-support.md (hunk 5 above)
Comment (Chriscbr) [reply docs/07-faq/023-kubernetes-support.md:15]: Sure, either sounds ok

File: docs/07-faq/023-kubernetes-support.md (hunk 4 above)
Comment (staycoolcall911) [reply docs/07-faq/023-kubernetes-support.md]: > IDE autocomplete doesn't work after you type new kube.deployment.

Just wanted to add that this autocomplete bug is not specific to Kubernetes. It's on Mark's radar and should be fixed soon.
https://github.com/winglang/wing/issues/2339

**Review by shaiber01 — APPROVED** [review]

**Pull request merged**